client/start.go

```diff
@@ -35,8 +35,12 @@ const (
 	stderrFile = "serverlog_stderr.txt"
 )
 
-func start(c string) (func(), error) {
+func start(c string, runInContainer bool) (func(), error) {
 	args := strings.Fields(c)
+	// If run in a container, -cmd is the name of the image.
+	if runInContainer {
+		args = []string{"docker", "run", "--network=host", c}
```

Comment:
Can you run it in the background? `--detach`, output should print the container id that you can store for later and you wouldn't need to worry about process cleanup.

Reply:
I think docker starts a child process to run the container and it's not killed by in the cleanup code with or without `--detach`.